### PR TITLE
Add traceability to external requirements in feature files

### LINKED ISSUE #13: Provide traceability to external requirements
> As a modeler I want to show traceability in my model to external customer requirements (i.e. not captured in the AaC model) > so that my customer can see how my system model addresses needs.
>
> AC:
>
> 1. Must create a simple representation of a requirements spec (i.e. CSV file - not unlike a DOORS export)
> 2. Must extend the model language (either directly or through extension) to enable trace from enum, data, and model items > (model may actually be at the behavior level...or maybe both)

As an AaC Developer, I want to allow requirement IDs to be mapped for referencing, so that end users know if a model, use case behavior, or Gherkin feature file, assists with meeting a requirement for traceability. 

AC:

- [x] Requirement ID to behavior scenario can be mapped to model.
- [x] Requirement ID to model can be mapped.
- [x] Requirement ID to feature file can be mapped to behavior scenario.
- [x] Field for capturing requirement ID for mapping is added.
- [x] Update documentation to explain the role and usage of requirement ID

### PR DESCRIPTION
**Pull request by cvchaparro**
# Description

Map requirements to feature files.

# Linked Items:

Closes #13

### Changed

- Update the `gen-gherkin-behaviors` plugin to pass requirements along to generated feature files.

# Checklist:

- [ ] I updated project documentation to reflect my changes.
- [x] My changes generate no new warnings.
- [x] I updated new and existing unit tests to account for my changes.
- [x] I linked the associated item(s) to be closed.
- [x] I bumped the version.
- [x] I added the labels corresponding to my changes.

### CODE REVIEW

**Comment by lizzcondrey** [comment]
Wooo can't hit unit tests still we sort out this artifact build mess. But running tox within GitPod gives me back this. Find it interesting that we get an error for pygls and that's the package seeming to cause issues in the pipeline.

```
======================================================================
ERROR: test_handles_goto_definition_request_when_definition_in_different_document (tests.plugins.lsp_server.providers.test_goto_definition_provider.TestGotoDefinitionProvider)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python3.9/unittest/async_case.py", line 64, in _callTestMethod
    self._callMaybeAsync(method)
  File "/usr/lib/python3.9/unittest/async_case.py", line 87, in _callMaybeAsync
    return self._asyncioTestLoop.run_until_complete(fut)
  File "/usr/lib/python3.9/asyncio/base_events.py", line 647, in run_until_complete
    return future.result()
  File "/usr/lib/python3.9/unittest/async_case.py", line 101, in _asyncioLoopRunner
    ret = await awaitable
  File "/workspace/AaC/python/tests/plugins/lsp_server/providers/test_goto_definition_provider.py", line 114, in test_handles_goto_definition_request_when_definition_in_different_document
    res: GotoDefinitionResponse = await self.goto_definition(TEST_DOCUMENT_NAME, line=61, character=17)
  File "/workspace/AaC/python/tests/plugins/lsp_server/providers/test_goto_definition_provider.py", line 63, in goto_definition
    return await self.build_request(
  File "/workspace/AaC/python/tests/plugins/lsp_server/base_lsp_test_case.py", line 186, in build_request
    return response_type(response.result())
  File "/usr/lib/python3.9/concurrent/futures/_base.py", line 439, in result
    return self.__get_result()
  File "/usr/lib/python3.9/concurrent/futures/_base.py", line 391, in __get_result
    raise self._exception
pygls.exceptions.JsonRpcInternalError: IndexError: string index out of range

======================================================================
FAIL: test_get_diagnostics_for_document_with_valid_definitions (tests.plugins.lsp_server.providers.test_publish_diagnostics_provider.TestPublishDiagnosticsProvider)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python3.9/unittest/async_case.py", line 64, in _callTestMethod
    self._callMaybeAsync(method)
  File "/usr/lib/python3.9/unittest/async_case.py", line 87, in _callMaybeAsync
    return self._asyncioTestLoop.run_until_complete(fut)
  File "/usr/lib/python3.9/asyncio/base_events.py", line 647, in run_until_complete
    return future.result()
  File "/usr/lib/python3.9/unittest/async_case.py", line 101, in _asyncioLoopRunner
    ret = await awaitable
  File "/workspace/AaC/python/tests/plugins/lsp_server/providers/test_publish_diagnostics_provider.py", line 36, in test_get_diagnostics_for_document_with_valid_definitions
    self.assertListEqual([], error_diagnostics)
AssertionError: Lists differ: [] != [Diagnostic(range=14:0-14:5, message="Requ[205 chars]one)]

Second list contains 1 additional elements.
First extra element 0:
Diagnostic(range=14:0-14:5, message="Required field 'ids' is not populated in: {'ids': []}", severity=<DiagnosticSeverity.Error: 1>, code='Validate required fields', code_description=None, source='aac', related_information=None, tags=None, data=None)

- []
+ [Diagnostic(range=14:0-14:5, message="Required field 'ids' is not populated in: {'ids': []}", severity=<DiagnosticSeverity.Error: 1>, code='Validate required fields', code_description=None, source='aac', related_information=None, tags=None, data=None)]

======================================================================
FAIL: test_parser_builds_node_structure_model (tests.lang.definitions.test_metadata_structure.TestDefinitionNodeStructure)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/workspace/AaC/python/tests/lang/definitions/test_metadata_structure.py", line 100, in test_parser_builds_node_structure_model
    _assert_structural_node(self, behaviors_entry1_node, NodeType.LIST_ENTRY, None, 7)
  File "/workspace/AaC/python/tests/lang/definitions/test_metadata_structure.py", line 168, in _assert_structural_node
    test_case.assertEqual(expected_child_count, len(actual_node.children))
AssertionError: 7 != 8

----------------------------------------------------------------------
Ran 351 tests in 379.082s

FAILED (failures=2, errors=1)
```

**Comment by cvchaparro** [comment]
> Wooo can't hit unit tests still we sort out this artifact build mess. But running tox within GitPod gives me back this. Find it interesting that we get an error for pygls and that's the package seeming to cause issues in the pipeline.
> 
> ```
> ======================================================================
> ERROR: test_handles_goto_definition_request_when_definition_in_different_document (tests.plugins.lsp_server.providers.test_goto_definition_provider.TestGotoDefinitionProvider)
> ----------------------------------------------------------------------
> Traceback (most recent call last):
>   File "/usr/lib/python3.9/unittest/async_case.py", line 64, in _callTestMethod
>     self._callMaybeAsync(method)
>   File "/usr/lib/python3.9/unittest/async_case.py", line 87, in _callMaybeAsync
>     return self._asyncioTestLoop.run_until_complete(fut)
>   File "/usr/lib/python3.9/asyncio/base_events.py", line 647, in run_until_complete
>     return future.result()
>   File "/usr/lib/python3.9/unittest/async_case.py", line 101, in _asyncioLoopRunner
>     ret = await awaitable
>   File "/workspace/AaC/python/tests/plugins/lsp_server/providers/test_goto_definition_provider.py", line 114, in test_handles_goto_definition_request_when_definition_in_different_document
>     res: GotoDefinitionResponse = await self.goto_definition(TEST_DOCUMENT_NAME, line=61, character=17)
>   File "/workspace/AaC/python/tests/plugins/lsp_server/providers/test_goto_definition_provider.py", line 63, in goto_definition
>     return await self.build_request(
>   File "/workspace/AaC/python/tests/plugins/lsp_server/base_lsp_test_case.py", line 186, in build_request
>     return response_type(response.result())
>   File "/usr/lib/python3.9/concurrent/futures/_base.py", line 439, in result
>     return self.__get_result()
>   File "/usr/lib/python3.9/concurrent/futures/_base.py", line 391, in __get_result
>     raise self._exception
> pygls.exceptions.JsonRpcInternalError: IndexError: string index out of range
> 
> ======================================================================
> FAIL: test_get_diagnostics_for_document_with_valid_definitions (tests.plugins.lsp_server.providers.test_publish_diagnostics_provider.TestPublishDiagnosticsProvider)
> ----------------------------------------------------------------------
> Traceback (most recent call last):
>   File "/usr/lib/python3.9/unittest/async_case.py", line 64, in _callTestMethod
>     self._callMaybeAsync(method)
>   File "/usr/lib/python3.9/unittest/async_case.py", line 87, in _callMaybeAsync
>     return self._asyncioTestLoop.run_until_complete(fut)
>   File "/usr/lib/python3.9/asyncio/base_events.py", line 647, in run_until_complete
>     return future.result()
>   File "/usr/lib/python3.9/unittest/async_case.py", line 101, in _asyncioLoopRunner
>     ret = await awaitable
>   File "/workspace/AaC/python/tests/plugins/lsp_server/providers/test_publish_diagnostics_provider.py", line 36, in test_get_diagnostics_for_document_with_valid_definitions
>     self.assertListEqual([], error_diagnostics)
> AssertionError: Lists differ: [] != [Diagnostic(range=14:0-14:5, message="Requ[205 chars]one)]
> 
> Second list contains 1 additional elements.
> First extra element 0:
> Diagnostic(range=14:0-14:5, message="Required field 'ids' is not populated in: {'ids': []}", severity=<DiagnosticSeverity.Error: 1>, code='Validate required fields', code_description=None, source='aac', related_information=None, tags=None, data=None)
> 
> - []
> + [Diagnostic(range=14:0-14:5, message="Required field 'ids' is not populated in: {'ids': []}", severity=<DiagnosticSeverity.Error: 1>, code='Validate required fields', code_description=None, source='aac', related_information=None, tags=None, data=None)]
> 
> ======================================================================
> FAIL: test_parser_builds_node_structure_model (tests.lang.definitions.test_metadata_structure.TestDefinitionNodeStructure)
> ----------------------------------------------------------------------
> Traceback (most recent call last):
>   File "/workspace/AaC/python/tests/lang/definitions/test_metadata_structure.py", line 100, in test_parser_builds_node_structure_model
>     _assert_structural_node(self, behaviors_entry1_node, NodeType.LIST_ENTRY, None, 7)
>   File "/workspace/AaC/python/tests/lang/definitions/test_metadata_structure.py", line 168, in _assert_structural_node
>     test_case.assertEqual(expected_child_count, len(actual_node.children))
> AssertionError: 7 != 8
> 
> ----------------------------------------------------------------------
> Ran 351 tests in 379.082s
> 
> FAILED (failures=2, errors=1)
> ```

Hey, @lizzcondrey - thanks for pointing this out! Apparently I didn't run the full test suite before. These tests should be fixed in commit 812fb00

File: python/tests/helpers/parsed_definitions.py
Comment (lizzcondrey) [inline python/tests/helpers/parsed_definitions.py:1]: Yeap, the tweaks here helped fixed the failing tests! Good job!

**Review by lizzcondrey — APPROVED** [review]
Looks good. Output with the requirements IDs should be a great add for users!

**Review by lizzcondrey — APPROVED** [review]
Need to complete the review of my code snippet branch so you get the corrected workflow on remerge of main.

**Comment by lizzcondrey** [comment]
@cvchaparro got my branch merged in, so you should be good to pull in main to fix the workflow break and get this branch merged in.

**Pull request merged**